src/main/java/triss/Triss.java

```diff
@@ -0,0 +1,429 @@
+                    + " \n"
+                    + "Try inserting a deadline in this format:\n"
+                    + "    deadline Meet with Friends /12th July";
+            throw new TrissException(errorMessage);
+        }
+
+        // Catch other possible errors
+        // Throw error if user did not type in a name for the task
+        if (taskName.isBlank()) {
+            throw new TrissException("Your deadline name is blank! Let's try that again.");
+        }
+
+        // Throw error if user did not type in a timing for the event
+        if (deadlineDate.isBlank()) {
+            throw new TrissException("You didn't insert a date in your deadline! Let's try that again.");
+        }
+
+        // Add deadline to tasks
+        Deadline newDeadline = new Deadline(taskName, deadlineDate);
+        tasks.add(newDeadline);
+
+        // Then, echo the task if not silent
+        if (!isSilent) {
+            printLine("I've added: " + newDeadline.printTask());
+        }
+    }
+
+    /**
+     * Mark user task as done, if request is valid.
+     * Stops if user did not specify a task.
+     * Stops if user's chosen task does not exist.
+     * Informs user if task was already done.
+     * @param userInput Any user input starting with "done"
+     */
+    private static void handleUserMarkingTaskAsDone(String userInput) {
+        // Get number of task after the term "done"
+        int indexOfCompletedTask;
+        // Throw exception if user did not type a number after "done"
+        try {
+            indexOfCompletedTask = Integer.parseInt(parseUserInput(userInput, 1)) - 1;
+        } catch (Exception e) {
+            printLine("Ach, nee! That task does not exist.");
+            return;
+        }
+
+
+        // If task does not exist, do not delete any task
+        if (indexOfCompletedTask >= tasks.size() || indexOfCompletedTask < 0) {
+            printLine("Apologies! That task does not exist.");
+            return;
+        }
+
+        // Find task since it exists
+        Task chosenTask = tasks.get(indexOfCompletedTask);
+
+        // If task was already done, let user know
+        if (chosenTask.isDone()) {
+            printLine("Oh! This task was already marked as done:");
+            // Print out the task in the following format: "    [X] Task"
+            printLine("    " + chosenTask.printTask());
+            return;
+        }
+
+        // If task exists, and is not done, mark it as done
+        chosenTask.setDone(true);
+        printLine("Wunderbar! This task has been marked as done:");
+
+        // Print out the task in the following format: "    [X] Task"
+        printLine("    " + chosenTask.printTask());
+    }
+
+    /**
+     * Prints all tasks stored in Task Array tasks.
+     */
+    private static void printAllTasks() {
+        // If user said "list", print a list of all saved tasks
+        for (Task task:tasks) {
```

Comment:
Minor nitpick but you could add spaces between the `:` for slightly better readability.

Reply:
Roger that.